diff --git a/src/qabstract_item_model/proxy_rust.rs b/src/qabstract_item_model/proxy_rust.rs
index 39d7e8b46..ed3cf0ea8 100644
--- a/src/qabstract_item_model/proxy_rust.rs
+++ b/src/qabstract_item_model/proxy_rust.rs
@@ -63,11 +63,11 @@ pub trait QAbstractItemModel : QObjectHolder<ProxyRust = QAbstractItemModelProxy
     }
     fn set_data(&mut self, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
         let proxy = self.try_get_rust_proxy_ptr().expect("No proxy");
-        unsafe { &mut *proxy }.base_set_data(&mut *self, index, value, role)
+        unsafe { &*proxy }.base_set_data(&mut *self, index, value, role)
     }
     fn remove_rows(&mut self, first: i32, count: i32, parent: &QModelIndex) -> bool {
         let proxy = self.try_get_rust_proxy_ptr().expect("No proxy");
-        unsafe { &mut *proxy }.base_remove_rows(&mut *self, first, count, parent)
+        unsafe { &*proxy }.base_remove_rows(&mut *self, first, count, parent)
     }
     fn sibling(&self, row: i32, column: i32, idx: &QModelIndex) -> QModelIndex {
         let proxy = self.try_get_rust_proxy_ptr().expect("No proxy");
@@ -87,7 +87,7 @@ pub trait QAbstractItemModelBase : QAbstractItemModel {
         role: i32,
     ) -> bool {
         let proxy = self.try_get_rust_proxy_ptr().expect("No proxy");
-        unsafe { &mut *proxy }.base_set_data(&mut *self, index, value, role)
+        unsafe { &*proxy }.base_set_data(&mut *self, index, value, role)
     }
     fn remove_rows(
         &mut self,
@@ -96,7 +96,7 @@ pub trait QAbstractItemModelBase : QAbstractItemModel {
         parent: &QModelIndex,
     ) -> bool {
         let proxy = self.try_get_rust_proxy_ptr().expect("No proxy");
-        unsafe { &mut *proxy }.base_remove_rows(&mut *self, first, count, parent)
+        unsafe { &*proxy }.base_remove_rows(&mut *self, first, count, parent)
     }
     fn sibling(
         &self,
@@ -113,7 +113,7 @@ pub trait QAbstractItemModelBase : QAbstractItemModel {
         bottom_right: &QModelIndex,
     ) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_data_changed(&mut *self, top_left, bottom_right)
+        unsafe { &*proxy }.base_data_changed(&mut *self, top_left, bottom_right)
     }
     fn begin_insert_columns(
         &mut self,
@@ -122,11 +122,11 @@ pub trait QAbstractItemModelBase : QAbstractItemModel {
         last: i32,
     ) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_begin_insert_columns(&mut *self, parent, first, last)
+        unsafe { &*proxy }.base_begin_insert_columns(&mut *self, parent, first, last)
     }
     fn end_insert_columns(&mut self) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_end_insert_columns(&mut *self)
+        unsafe { &*proxy }.base_end_insert_columns(&mut *self)
     }
     fn begin_insert_rows(
         &mut self,
@@ -135,11 +135,11 @@ pub trait QAbstractItemModelBase : QAbstractItemModel {
         last: i32,
     ) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_begin_insert_rows(&mut *self, parent, first, last)
+        unsafe { &*proxy }.base_begin_insert_rows(&mut *self, parent, first, last)
     }
     fn end_insert_rows(&mut self) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_end_insert_rows(&mut *self)
+        unsafe { &*proxy }.base_end_insert_rows(&mut *self)
     }
     fn begin_move_columns(
         &mut self,
@@ -150,7 +150,7 @@ pub trait QAbstractItemModelBase : QAbstractItemModel {
         destination_child: i32,
     ) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_begin_move_columns(
+        unsafe { &*proxy }.base_begin_move_columns(
             &mut *self,
             source_parent,
             source_first,
@@ -161,7 +161,7 @@ pub trait QAbstractItemModelBase : QAbstractItemModel {
     }
     fn end_move_columns(&mut self) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_end_move_columns(&mut *self)
+        unsafe { &*proxy }.base_end_move_columns(&mut *self)
     }
     fn begin_move_rows(
         &mut self,
@@ -172,7 +172,7 @@ pub trait QAbstractItemModelBase : QAbstractItemModel {
         destination_child: i32,
     ) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_begin_move_rows(
+        unsafe { &*proxy }.base_begin_move_rows(
             &mut *self,
             source_parent,
             source_first,
@@ -183,7 +183,7 @@ pub trait QAbstractItemModelBase : QAbstractItemModel {
     }
     fn end_move_rows(&mut self) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_end_move_rows(&mut *self)
+        unsafe { &*proxy }.base_end_move_rows(&mut *self)
     }
     fn begin_remove_columns(
         &mut self,
@@ -192,11 +192,11 @@ pub trait QAbstractItemModelBase : QAbstractItemModel {
         last: i32,
     ) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_begin_remove_columns(&mut *self, parent, first, last)
+        unsafe { &*proxy }.base_begin_remove_columns(&mut *self, parent, first, last)
     }
     fn end_remove_columns(&mut self) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_end_remove_columns(&mut *self)
+        unsafe { &*proxy }.base_end_remove_columns(&mut *self)
     }
     fn begin_remove_rows(
         &mut self,
@@ -205,19 +205,19 @@ pub trait QAbstractItemModelBase : QAbstractItemModel {
         last: i32,
     ) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_begin_remove_rows(&mut *self, parent, first, last)
+        unsafe { &*proxy }.base_begin_remove_rows(&mut *self, parent, first, last)
     }
     fn end_remove_rows(&mut self) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_end_remove_rows(&mut *self)
+        unsafe { &*proxy }.base_end_remove_rows(&mut *self)
     }
     fn begin_reset_model(&mut self) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_begin_reset_model(&mut *self)
+        unsafe { &*proxy }.base_begin_reset_model(&mut *self)
     }
     fn end_reset_model(&mut self) {
         let Some(proxy) = self.try_get_rust_proxy_ptr() else { return };
-        unsafe { &mut *proxy }.base_end_reset_model(&mut *self)
+        unsafe { &*proxy }.base_end_reset_model(&mut *self)
     }
     fn create_index(
         &self,
@@ -260,10 +260,10 @@ impl QAbstractItemModelProxyRust {
     pub fn role_names(&self) -> QHash_i32_QByteArray {
         call_rust_trait_impl!(self, role_names())
     }
-    pub fn set_data(&mut self, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
+    pub fn set_data(&self, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
         call_rust_trait_impl!(mut self, set_data(index, value, role))
     }
-    pub fn remove_rows(&mut self, first: i32, count: i32, parent: &QModelIndex) -> bool {
+    pub fn remove_rows(&self, first: i32, count: i32, parent: &QModelIndex) -> bool {
         call_rust_trait_impl!(mut self, remove_rows(first, count, parent))
     }
     pub fn sibling(&self, row: i32, column: i32, idx: &QModelIndex) -> QModelIndex {
@@ -273,32 +273,32 @@ impl QAbstractItemModelProxyRust {
     pub fn base_role_names(&self, reference: &dyn QAbstractItemModelAdapter) -> QHash_i32_QByteArray {
         call_cpp_impl!(self, reference, base_role_names())
     }
-    pub fn base_set_data(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
+    pub fn base_set_data(&self, mut_ref: &mut dyn QAbstractItemModelAdapter, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
         call_cpp_impl!(mut self, mut_ref, base_set_data(index, value, role))
     }
-    pub fn base_remove_rows(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter, first: i32, count: i32, parent: &QModelIndex) -> bool {
+    pub fn base_remove_rows(&self, mut_ref: &mut dyn QAbstractItemModelAdapter, first: i32, count: i32, parent: &QModelIndex) -> bool {
         call_cpp_impl!(mut self, mut_ref, base_remove_rows(first, count, parent))
     }
     pub fn base_sibling(&self, reference: &dyn QAbstractItemModelAdapter, row: i32, column: i32, idx: &QModelIndex) -> QModelIndex {
         call_cpp_impl!(self, reference, base_sibling(row, column, idx))
     }
-    pub fn base_data_changed(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter, top_left: &QModelIndex, bottom_right: &QModelIndex) {
+    pub fn base_data_changed(&self, mut_ref: &mut dyn QAbstractItemModelAdapter, top_left: &QModelIndex, bottom_right: &QModelIndex) {
         call_cpp_impl!(mut self, mut_ref, base_data_changed(top_left, bottom_right))
     }
-    pub fn base_begin_insert_columns(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
+    pub fn base_begin_insert_columns(&self, mut_ref: &mut dyn QAbstractItemModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
         call_cpp_impl!(mut self, mut_ref, base_begin_insert_columns(parent, first, last))
     }
-    pub fn base_end_insert_columns(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
+    pub fn base_end_insert_columns(&self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_insert_columns())
     }
-    pub fn base_begin_insert_rows(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
+    pub fn base_begin_insert_rows(&self, mut_ref: &mut dyn QAbstractItemModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
         call_cpp_impl!(mut self, mut_ref, base_begin_insert_rows(parent, first, last))
     }
-    pub fn base_end_insert_rows(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
+    pub fn base_end_insert_rows(&self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_insert_rows())
     }
     pub fn base_begin_move_columns(
-        &mut self,
+        &self,
         mut_ref: &mut dyn QAbstractItemModelAdapter,
         source_parent: &QModelIndex,
         source_first: i32,
@@ -308,11 +308,11 @@ impl QAbstractItemModelProxyRust {
     ) {
         call_cpp_impl!(mut self, mut_ref, base_begin_move_columns(source_parent, source_first, source_last, destination_parent, destination_child))
     }
-    pub fn base_end_move_columns(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
+    pub fn base_end_move_columns(&self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_move_columns())
     }
     pub fn base_begin_move_rows(
-        &mut self,
+        &self,
         mut_ref: &mut dyn QAbstractItemModelAdapter,
         source_parent: &QModelIndex,
         source_first: i32,
@@ -322,25 +322,25 @@ impl QAbstractItemModelProxyRust {
     ) {
         call_cpp_impl!(mut self, mut_ref, base_begin_move_rows(source_parent, source_first, source_last, destination_parent, destination_child))
     }
-    pub fn base_end_move_rows(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
+    pub fn base_end_move_rows(&self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_move_rows())
     }
-    pub fn base_begin_remove_columns(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
+    pub fn base_begin_remove_columns(&self, mut_ref: &mut dyn QAbstractItemModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
         call_cpp_impl!(mut self, mut_ref, base_begin_remove_columns(parent, first, last))
     }
-    pub fn base_end_remove_columns(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
+    pub fn base_end_remove_columns(&self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_remove_columns())
     }
-    pub fn base_begin_remove_rows(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
+    pub fn base_begin_remove_rows(&self, mut_ref: &mut dyn QAbstractItemModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
         call_cpp_impl!(mut self, mut_ref, base_begin_remove_rows(parent, first, last))
     }
-    pub fn base_end_remove_rows(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
+    pub fn base_end_remove_rows(&self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_remove_rows())
     }
-    pub fn base_begin_reset_model(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
+    pub fn base_begin_reset_model(&self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_begin_reset_model())
     }
-    pub fn base_end_reset_model(&mut self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
+    pub fn base_end_reset_model(&self, mut_ref: &mut dyn QAbstractItemModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_reset_model())
     }
     pub fn base_create_index(&self, reference: &dyn QAbstractItemModelAdapter, row: i32, column: i32, ptr: usize) -> QModelIndex {
diff --git a/src/qabstract_item_model/proxy_rust_bridge.rs b/src/qabstract_item_model/proxy_rust_bridge.rs
index a0477331d..3a4a3eb7e 100644
--- a/src/qabstract_item_model/proxy_rust_bridge.rs
+++ b/src/qabstract_item_model/proxy_rust_bridge.rs
@@ -30,9 +30,9 @@ pub mod ffi {
         # [cxx_name = roleNames]
         fn role_names(&self) -> QHash_i32_QByteArray;
         # [cxx_name = setData]
-        fn set_data(&mut self, index: &QModelIndex, value: &QVariant, role: i32) -> bool;
+        fn set_data(&self, index: &QModelIndex, value: &QVariant, role: i32) -> bool;
         # [cxx_name = removeRows]
-        fn remove_rows(&mut self, first: i32, count: i32, parent: &QModelIndex) -> bool;
+        fn remove_rows(&self, first: i32, count: i32, parent: &QModelIndex) -> bool;
         # [cxx_name = sibling]
         fn sibling(&self, row: i32, column: i32, idx: &QModelIndex) -> QModelIndex;
 
diff --git a/src/qlist_model/proxy_rust.rs b/src/qlist_model/proxy_rust.rs
index 28be5cf87..138456ee2 100644
--- a/src/qlist_model/proxy_rust.rs
+++ b/src/qlist_model/proxy_rust.rs
@@ -61,7 +61,7 @@ where
         if updated {
             self.set_unnotified(index.row() as usize, item);
             let proxy = self.try_get_rust_proxy_ptr().expect("No proxy");
-            unsafe { &mut *proxy }.base_data_changed(&mut *self, index, index);
+            unsafe { &*proxy }.base_data_changed(&mut *self, index, index);
         }
         updated
     }
@@ -73,11 +73,11 @@ where
             return false;
         }
         let proxy = self.try_get_rust_proxy_ptr().expect("No proxy");
-        unsafe { &mut *proxy }.base_begin_remove_rows(&mut *self, parent, first as i32, (last - 1) as i32);
+        unsafe { &*proxy }.base_begin_remove_rows(&mut *self, parent, first as i32, (last - 1) as i32);
         for index in (first..last).rev() {
             self.remove_unnotified(index);
         }
-        unsafe { &mut *proxy }.base_end_remove_rows(&mut *self);
+        unsafe { &*proxy }.base_end_remove_rows(&mut *self);
         true
     }
 
@@ -291,7 +291,7 @@ pub trait QListModelBase : QListModel + QObjectHolder<ProxyRust = QListModelProx
         if self.set_unnotified(index, value) {
             if let Some(proxy) = self.try_get_rust_proxy_ptr() {
                 let model_index = unsafe { &*proxy }.base_index(&*self, index as i32, 0, &QModelIndex::default());
-                unsafe { &mut *proxy }.base_data_changed(&mut *self, &model_index, &model_index);
+                unsafe { &*proxy }.base_data_changed(&mut *self, &model_index, &model_index);
             }
             true
         } else {
@@ -308,9 +308,9 @@ pub trait QListModelBase : QListModel + QObjectHolder<ProxyRust = QListModelProx
             return self.push_unnotified(value);
         };
         let len = self.len() as i32;
-        unsafe { &mut *proxy }.base_begin_insert_rows(&mut *self, &QModelIndex::default(), len, len);
+        unsafe { &*proxy }.base_begin_insert_rows(&mut *self, &QModelIndex::default(), len, len);
         self.push_unnotified(value);
-        unsafe { &mut *proxy }.base_end_insert_rows(&mut *self);
+        unsafe { &*proxy }.base_end_insert_rows(&mut *self);
     }
 
     /// Inserts `value` at `index` and notifies any attached views about
@@ -321,9 +321,9 @@ pub trait QListModelBase : QListModel + QObjectHolder<ProxyRust = QListModelProx
         let Some(proxy) = self.try_get_rust_proxy_ptr() else {
             return self.insert_unnotified(index, value);
         };
-        unsafe { &mut *proxy }.base_begin_insert_rows(&mut *self, &QModelIndex::default(), index as i32, index as i32);
+        unsafe { &*proxy }.base_begin_insert_rows(&mut *self, &QModelIndex::default(), index as i32, index as i32);
         self.insert_unnotified(index, value);
-        unsafe { &mut *proxy }.base_end_insert_rows(&mut *self);
+        unsafe { &*proxy }.base_end_insert_rows(&mut *self);
     }
 
     /// Removes and returns the last item in the model and notifies any attached views about
@@ -341,9 +341,9 @@ pub trait QListModelBase : QListModel + QObjectHolder<ProxyRust = QListModelProx
             return self.pop_unnotified();
         };
         let len = self.len() as i32;
-        unsafe { &mut *proxy }.base_begin_remove_rows(&mut *self, &QModelIndex::default(), len - 1, len - 1);
+        unsafe { &*proxy }.base_begin_remove_rows(&mut *self, &QModelIndex::default(), len - 1, len - 1);
         let value = self.pop_unnotified();
-        unsafe { &mut *proxy }.base_end_remove_rows(&mut *self);
+        unsafe { &*proxy }.base_end_remove_rows(&mut *self);
         value
     }
 
@@ -355,9 +355,9 @@ pub trait QListModelBase : QListModel + QObjectHolder<ProxyRust = QListModelProx
         let Some(proxy) = self.try_get_rust_proxy_ptr() else {
             return self.remove_unnotified(index);
         };
-        unsafe { &mut *proxy }.base_begin_remove_rows(&mut *self, &QModelIndex::default(), index as i32, index as i32);
+        unsafe { &*proxy }.base_begin_remove_rows(&mut *self, &QModelIndex::default(), index as i32, index as i32);
         let value = self.remove_unnotified(index);
-        unsafe { &mut *proxy }.base_end_remove_rows(&mut *self);
+        unsafe { &*proxy }.base_end_remove_rows(&mut *self);
         value
     }
 
@@ -368,9 +368,9 @@ pub trait QListModelBase : QListModel + QObjectHolder<ProxyRust = QListModelProx
         let Some(proxy) = self.try_get_rust_proxy_ptr() else {
             return self.reset_unnotified();
         };
-        unsafe { &mut *proxy }.base_begin_reset_model(&mut *self);
+        unsafe { &*proxy }.base_begin_reset_model(&mut *self);
         self.reset_unnotified();
-        unsafe { &mut *proxy }.base_end_reset_model(&mut *self);
+        unsafe { &*proxy }.base_end_reset_model(&mut *self);
     }
 }
 
@@ -398,10 +398,10 @@ impl QListModelProxyRust {
     pub fn role_names(&self) -> QHash_i32_QByteArray {
         call_rust_trait_impl!(self, role_names())
     }
-    pub fn set_data(&mut self, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
+    pub fn set_data(&self, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
         call_rust_trait_impl!(mut self, set_data(index, value, role))
     }
-    pub fn remove_rows(&mut self, first: i32, count: i32, parent: &QModelIndex) -> bool {
+    pub fn remove_rows(&self, first: i32, count: i32, parent: &QModelIndex) -> bool {
         call_rust_trait_impl!(mut self, remove_rows(first, count, parent))
     }
     pub fn sibling(&self, row: i32, column: i32, idx: &QModelIndex) -> QModelIndex {
@@ -414,40 +414,40 @@ impl QListModelProxyRust {
     pub fn base_role_names(&self, reference: &dyn QListModelAdapter) -> QHash_i32_QByteArray {
         call_cpp_impl!(self, reference, base_role_names())
     }
-    pub fn base_set_data(&mut self, mut_ref: &mut dyn QListModelAdapter, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
+    pub fn base_set_data(&self, mut_ref: &mut dyn QListModelAdapter, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
         call_cpp_impl!(mut self, mut_ref, base_set_data(index, value, role))
     }
-    pub fn base_remove_rows(&mut self, mut_ref: &mut dyn QListModelAdapter, first: i32, count: i32, parent: &QModelIndex) -> bool {
+    pub fn base_remove_rows(&self, mut_ref: &mut dyn QListModelAdapter, first: i32, count: i32, parent: &QModelIndex) -> bool {
         call_cpp_impl!(mut self, mut_ref, base_remove_rows(first, count, parent))
     }
     pub fn base_sibling(&self, reference: &dyn QListModelAdapter, row: i32, column: i32, idx: &QModelIndex) -> QModelIndex {
         call_cpp_impl!(self, reference, base_sibling(row, column, idx))
     }
-    pub fn base_data_changed(&mut self, mut_ref: &mut dyn QListModelAdapter, top_left: &QModelIndex, bottom_right: &QModelIndex) {
+    pub fn base_data_changed(&self, mut_ref: &mut dyn QListModelAdapter, top_left: &QModelIndex, bottom_right: &QModelIndex) {
         call_cpp_impl!(mut self, mut_ref, base_data_changed(top_left, bottom_right))
     }
-    pub fn base_begin_insert_rows(&mut self, mut_ref: &mut dyn QListModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
+    pub fn base_begin_insert_rows(&self, mut_ref: &mut dyn QListModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
         call_cpp_impl!(mut self, mut_ref, base_begin_insert_rows(parent, first, last))
     }
-    pub fn base_end_insert_rows(&mut self, mut_ref: &mut dyn QListModelAdapter) {
+    pub fn base_end_insert_rows(&self, mut_ref: &mut dyn QListModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_insert_rows())
     }
-    pub fn base_begin_move_rows(&mut self, mut_ref: &mut dyn QListModelAdapter, source_parent: &QModelIndex, source_first: i32, source_last: i32, destination_parent: &QModelIndex, destination_child: i32) {
+    pub fn base_begin_move_rows(&self, mut_ref: &mut dyn QListModelAdapter, source_parent: &QModelIndex, source_first: i32, source_last: i32, destination_parent: &QModelIndex, destination_child: i32) {
         call_cpp_impl!(mut self, mut_ref, base_begin_move_rows(source_parent, source_first, source_last, destination_parent, destination_child))
     }
-    pub fn base_end_move_rows(&mut self, mut_ref: &mut dyn QListModelAdapter) {
+    pub fn base_end_move_rows(&self, mut_ref: &mut dyn QListModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_move_rows())
     }
-    pub fn base_begin_remove_rows(&mut self, mut_ref: &mut dyn QListModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
+    pub fn base_begin_remove_rows(&self, mut_ref: &mut dyn QListModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
         call_cpp_impl!(mut self, mut_ref, base_begin_remove_rows(parent, first, last))
     }
-    pub fn base_end_remove_rows(&mut self, mut_ref: &mut dyn QListModelAdapter) {
+    pub fn base_end_remove_rows(&self, mut_ref: &mut dyn QListModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_remove_rows())
     }
-    pub fn base_begin_reset_model(&mut self, mut_ref: &mut dyn QListModelAdapter) {
+    pub fn base_begin_reset_model(&self, mut_ref: &mut dyn QListModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_begin_reset_model())
     }
-    pub fn base_end_reset_model(&mut self, mut_ref: &mut dyn QListModelAdapter) {
+    pub fn base_end_reset_model(&self, mut_ref: &mut dyn QListModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_reset_model())
     }
 }
diff --git a/src/qlist_model/proxy_rust_bridge.rs b/src/qlist_model/proxy_rust_bridge.rs
index c9e63ffab..ddecfd35c 100644
--- a/src/qlist_model/proxy_rust_bridge.rs
+++ b/src/qlist_model/proxy_rust_bridge.rs
@@ -26,9 +26,9 @@ pub mod ffi {
         # [cxx_name = roleNames]
         fn role_names(&self) -> QHash_i32_QByteArray;
         # [cxx_name = setData]
-        fn set_data(&mut self, index: &QModelIndex, value: &QVariant, role: i32) -> bool;
+        fn set_data(&self, index: &QModelIndex, value: &QVariant, role: i32) -> bool;
         # [cxx_name = removeRows]
-        fn remove_rows(&mut self, first: i32, count: i32, parent: &QModelIndex) -> bool;
+        fn remove_rows(&self, first: i32, count: i32, parent: &QModelIndex) -> bool;
         # [cxx_name = sibling]
         fn sibling(&self, row: i32, column: i32, idx: &QModelIndex) -> QModelIndex;
 
diff --git a/src/qparser_status/proxy_rust.rs b/src/qparser_status/proxy_rust.rs
index c4183d0cd..af152e27e 100644
--- a/src/qparser_status/proxy_rust.rs
+++ b/src/qparser_status/proxy_rust.rs
@@ -107,11 +107,11 @@ impl<T: QParserStatusAdapter + 'static> qtbridge_runtime::qproxies::AdapterUpcas
 }
 
 impl QParserStatusProxyRust {
-    pub fn class_begin(&mut self) {
+    pub fn class_begin(&self) {
         call_rust_trait_impl!(mut self, class_begin())
     }
 
-    pub fn component_complete(&mut self) {
+    pub fn component_complete(&self) {
         call_rust_trait_impl!(mut self, component_complete())
     }
 }
diff --git a/src/qparser_status/proxy_rust_bridge.rs b/src/qparser_status/proxy_rust_bridge.rs
index 8e713bc8f..31ec8a635 100644
--- a/src/qparser_status/proxy_rust_bridge.rs
+++ b/src/qparser_status/proxy_rust_bridge.rs
@@ -24,8 +24,8 @@ pub mod ffi {
         # [cxx_name = writeProperty]
         fn write_property(&self, prop_id: u32, value: &QVariant);
 
-        fn class_begin(&mut self);
-        fn component_complete(&mut self);
+        fn class_begin(&self);
+        fn component_complete(&self);
     }
 }
 unsafe impl cxx::ExternType for QParserStatusProxyRust {
diff --git a/src/qtable_model/proxy_rust.rs b/src/qtable_model/proxy_rust.rs
index e280cb61a..9e331e6f8 100644
--- a/src/qtable_model/proxy_rust.rs
+++ b/src/qtable_model/proxy_rust.rs
@@ -72,7 +72,7 @@ where
         if updated {
             self.set_unnotified((index.row() as usize, index.column() as usize), item);
             let proxy = self.try_get_rust_proxy_ptr().expect("No proxy");
-            unsafe { &mut *proxy }.base_data_changed(&mut *self, index, index);
+            unsafe { &*proxy }.base_data_changed(&mut *self, index, index);
         }
         updated
     }
@@ -84,11 +84,11 @@ where
             return false;
         }
         let proxy = self.try_get_rust_proxy_ptr().expect("No proxy");
-        unsafe { &mut *proxy }.base_begin_remove_columns(&mut *self, parent, first as i32, (last - 1) as i32);
+        unsafe { &*proxy }.base_begin_remove_columns(&mut *self, parent, first as i32, (last - 1) as i32);
         for index in (first..last).rev() {
             self.remove_column_unnotified(index);
         }
-        unsafe { &mut *proxy }.base_end_remove_columns(&mut *self);
+        unsafe { &*proxy }.base_end_remove_columns(&mut *self);
         true
     }
 
@@ -99,11 +99,11 @@ where
             return false;
         }
         let proxy = self.try_get_rust_proxy_ptr().expect("No proxy");
-        unsafe { &mut *proxy }.base_begin_remove_rows(&mut *self, parent, first as i32, (last - 1) as i32);
+        unsafe { &*proxy }.base_begin_remove_rows(&mut *self, parent, first as i32, (last - 1) as i32);
         for index in (first..last).rev() {
             self.remove_row_unnotified(index);
         }
-        unsafe { &mut *proxy }.base_end_remove_rows(&mut *self);
+        unsafe { &*proxy }.base_end_remove_rows(&mut *self);
         true
     }
 
@@ -378,7 +378,7 @@ pub trait QTableModelBase : QTableModel + QObjectHolder<ProxyRust = QTableModelP
         if self.set_unnotified(index, value) {
             if let Some(proxy) = self.try_get_rust_proxy_ptr() {
                 let model_index = unsafe { &*proxy }.base_create_index(self, index.0 as i32, index.1 as i32, 0);
-                unsafe { &mut *proxy }.base_data_changed(&mut *self, &model_index, &model_index);
+                unsafe { &*proxy }.base_data_changed(&mut *self, &model_index, &model_index);
             }
             true
         } else {
@@ -395,9 +395,9 @@ pub trait QTableModelBase : QTableModel + QObjectHolder<ProxyRust = QTableModelP
             return self.push_row_unnotified(values);
         };
         let row_count = self.row_count() as i32;
-        unsafe { &mut *proxy }.base_begin_insert_rows(&mut *self, &QModelIndex::default(), row_count, row_count);
+        unsafe { &*proxy }.base_begin_insert_rows(&mut *self, &QModelIndex::default(), row_count, row_count);
         self.push_row_unnotified(values);
-        unsafe { &mut *proxy }.base_end_insert_rows(&mut *self);
+        unsafe { &*proxy }.base_end_insert_rows(&mut *self);
     }
 
     /// Appends a column of `value` to the end of the model and notifies any attached views about
@@ -409,9 +409,9 @@ pub trait QTableModelBase : QTableModel + QObjectHolder<ProxyRust = QTableModelP
             return self.push_column_unnotified(values);
         };
         let col_count = self.column_count() as i32;
-        unsafe { &mut *proxy }.base_begin_insert_columns(&mut *self, &QModelIndex::default(), col_count, col_count);
+        unsafe { &*proxy }.base_begin_insert_columns(&mut *self, &QModelIndex::default(), col_count, col_count);
         self.push_column_unnotified(values);
-        unsafe { &mut *proxy }.base_end_insert_columns(&mut *self);
+        unsafe { &*proxy }.base_end_insert_columns(&mut *self);
     }
 
     /// Inserts a row with `values` at `index` and notifies any attached views about
@@ -422,9 +422,9 @@ pub trait QTableModelBase : QTableModel + QObjectHolder<ProxyRust = QTableModelP
         let Some(proxy) = self.try_get_rust_proxy_ptr() else {
             return self.insert_row_unnotified(index, values);
         };
-        unsafe { &mut *proxy }.base_begin_insert_rows(&mut *self, &QModelIndex::default(), index as i32, index as i32);
+        unsafe { &*proxy }.base_begin_insert_rows(&mut *self, &QModelIndex::default(), index as i32, index as i32);
         self.insert_row_unnotified(index, values);
-        unsafe { &mut *proxy }.base_end_insert_rows(&mut *self);
+        unsafe { &*proxy }.base_end_insert_rows(&mut *self);
     }
 
     /// Inserts a column with `values` at `index` and notifies any attached views about
@@ -435,9 +435,9 @@ pub trait QTableModelBase : QTableModel + QObjectHolder<ProxyRust = QTableModelP
         let Some(proxy) = self.try_get_rust_proxy_ptr() else {
             return self.insert_column_unnotified(index, values);
         };
-        unsafe { &mut *proxy }.base_begin_insert_columns(&mut *self, &QModelIndex::default(), index as i32, index as i32);
+        unsafe { &*proxy }.base_begin_insert_columns(&mut *self, &QModelIndex::default(), index as i32, index as i32);
         self.insert_column_unnotified(index, values);
-        unsafe { &mut *proxy }.base_end_insert_columns(&mut *self);
+        unsafe { &*proxy }.base_end_insert_columns(&mut *self);
     }
 
     /// Removes and returns the last row in the model and notifies any attached views about
@@ -455,9 +455,9 @@ pub trait QTableModelBase : QTableModel + QObjectHolder<ProxyRust = QTableModelP
             return self.pop_row_unnotified();
         };
         let row_count = self.row_count() as i32;
-        unsafe { &mut *proxy }.base_begin_remove_rows(&mut *self, &QModelIndex::default(), row_count - 1, row_count - 1);
+        unsafe { &*proxy }.base_begin_remove_rows(&mut *self, &QModelIndex::default(), row_count - 1, row_count - 1);
         let values = self.pop_row_unnotified();
-        unsafe { &mut *proxy }.base_end_remove_rows(&mut *self);
+        unsafe { &*proxy }.base_end_remove_rows(&mut *self);
         values
     }
 
@@ -476,9 +476,9 @@ pub trait QTableModelBase : QTableModel + QObjectHolder<ProxyRust = QTableModelP
             return self.pop_column_unnotified();
         };
         let col_count = self.column_count() as i32;
-        unsafe { &mut *proxy }.base_begin_remove_columns(&mut *self, &QModelIndex::default(), col_count - 1, col_count - 1);
+        unsafe { &*proxy }.base_begin_remove_columns(&mut *self, &QModelIndex::default(), col_count - 1, col_count - 1);
         let values = self.pop_column_unnotified();
-        unsafe { &mut *proxy }.base_end_remove_columns(&mut *self);
+        unsafe { &*proxy }.base_end_remove_columns(&mut *self);
         values
     }
 
@@ -490,9 +490,9 @@ pub trait QTableModelBase : QTableModel + QObjectHolder<ProxyRust = QTableModelP
         let Some(proxy) = self.try_get_rust_proxy_ptr() else {
             return self.remove_row_unnotified(index);
         };
-        unsafe { &mut *proxy }.base_begin_remove_rows(&mut *self, &QModelIndex::default(), index as i32, index as i32);
+        unsafe { &*proxy }.base_begin_remove_rows(&mut *self, &QModelIndex::default(), index as i32, index as i32);
         let values = self.remove_row_unnotified(index);
-        unsafe { &mut *proxy }.base_end_remove_rows(&mut *self);
+        unsafe { &*proxy }.base_end_remove_rows(&mut *self);
         values
     }
 
@@ -504,9 +504,9 @@ pub trait QTableModelBase : QTableModel + QObjectHolder<ProxyRust = QTableModelP
         let Some(proxy) = self.try_get_rust_proxy_ptr() else {
             return self.remove_column_unnotified(index);
         };
-        unsafe { &mut *proxy }.base_begin_remove_columns(&mut *self, &QModelIndex::default(), index as i32, index as i32);
+        unsafe { &*proxy }.base_begin_remove_columns(&mut *self, &QModelIndex::default(), index as i32, index as i32);
         let values = self.remove_column_unnotified(index);
-        unsafe { &mut *proxy }.base_end_remove_columns(&mut *self);
+        unsafe { &*proxy }.base_end_remove_columns(&mut *self);
         values
     }
 
@@ -517,9 +517,9 @@ pub trait QTableModelBase : QTableModel + QObjectHolder<ProxyRust = QTableModelP
         let Some(proxy) = self.try_get_rust_proxy_ptr() else {
             return self.reset_unnotified();
         };
-        unsafe { &mut *proxy }.base_begin_reset_model(&mut *self);
+        unsafe { &*proxy }.base_begin_reset_model(&mut *self);
         self.reset_unnotified();
-        unsafe { &mut *proxy }.base_end_reset_model(&mut *self);
+        unsafe { &*proxy }.base_end_reset_model(&mut *self);
     }
 }
 
@@ -553,13 +553,13 @@ impl QTableModelProxyRust {
     pub fn role_names(&self) -> QHash_i32_QByteArray {
         call_rust_trait_impl!(self, role_names())
     }
-    pub fn set_data(&mut self, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
+    pub fn set_data(&self, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
         call_rust_trait_impl!(mut self, set_data(index, value, role))
     }
-    pub fn remove_columns(&mut self, first: i32, count: i32, parent: &QModelIndex) -> bool {
+    pub fn remove_columns(&self, first: i32, count: i32, parent: &QModelIndex) -> bool {
         call_rust_trait_impl!(mut self, remove_columns(first, count, parent))
     }
-    pub fn remove_rows(&mut self, first: i32, count: i32, parent: &QModelIndex) -> bool {
+    pub fn remove_rows(&self, first: i32, count: i32, parent: &QModelIndex) -> bool {
         call_rust_trait_impl!(mut self, remove_rows(first, count, parent))
     }
     pub fn sibling(&self, row: i32, column: i32, idx: &QModelIndex) -> QModelIndex {
@@ -569,35 +569,35 @@ impl QTableModelProxyRust {
     pub fn base_role_names(&self, reference: &dyn QTableModelAdapter) -> QHash_i32_QByteArray {
         call_cpp_impl!(self, reference, base_role_names())
     }
-    pub fn base_set_data(&mut self, mut_ref: &mut dyn QTableModelAdapter, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
+    pub fn base_set_data(&self, mut_ref: &mut dyn QTableModelAdapter, index: &QModelIndex, value: &QVariant, role: i32) -> bool {
         call_cpp_impl!(mut self, mut_ref, base_set_data(index, value, role))
     }
-    pub fn base_remove_columns(&mut self, mut_ref: &mut dyn QTableModelAdapter, first: i32, count: i32, parent: &QModelIndex) -> bool {
+    pub fn base_remove_columns(&self, mut_ref: &mut dyn QTableModelAdapter, first: i32, count: i32, parent: &QModelIndex) -> bool {
         call_cpp_impl!(mut self, mut_ref, base_remove_columns(first, count, parent))
     }
-    pub fn base_remove_rows(&mut self, mut_ref: &mut dyn QTableModelAdapter, first: i32, count: i32, parent: &QModelIndex) -> bool {
+    pub fn base_remove_rows(&self, mut_ref: &mut dyn QTableModelAdapter, first: i32, count: i32, parent: &QModelIndex) -> bool {
         call_cpp_impl!(mut self, mut_ref, base_remove_rows(first, count, parent))
     }
     pub fn base_sibling(&self, reference: &dyn QTableModelAdapter, row: i32, column: i32, idx: &QModelIndex) -> QModelIndex {
         call_cpp_impl!(self, reference, base_sibling(row, column, idx))
     }
-    pub fn base_data_changed(&mut self, mut_ref: &mut dyn QTableModelAdapter, top_left: &QModelIndex, bottom_right: &QModelIndex) {
+    pub fn base_data_changed(&self, mut_ref: &mut dyn QTableModelAdapter, top_left: &QModelIndex, bottom_right: &QModelIndex) {
         call_cpp_impl!(mut self, mut_ref, base_data_changed(top_left, bottom_right))
     }
-    pub fn base_begin_insert_columns(&mut self, mut_ref: &mut dyn QTableModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
+    pub fn base_begin_insert_columns(&self, mut_ref: &mut dyn QTableModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
         call_cpp_impl!(mut self, mut_ref, base_begin_insert_columns(parent, first, last))
     }
-    pub fn base_end_insert_columns(&mut self, mut_ref: &mut dyn QTableModelAdapter) {
+    pub fn base_end_insert_columns(&self, mut_ref: &mut dyn QTableModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_insert_columns())
     }
-    pub fn base_begin_insert_rows(&mut self, mut_ref: &mut dyn QTableModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
+    pub fn base_begin_insert_rows(&self, mut_ref: &mut dyn QTableModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
         call_cpp_impl!(mut self, mut_ref, base_begin_insert_rows(parent, first, last))
     }
-    pub fn base_end_insert_rows(&mut self, mut_ref: &mut dyn QTableModelAdapter) {
+    pub fn base_end_insert_rows(&self, mut_ref: &mut dyn QTableModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_insert_rows())
     }
     pub fn base_begin_move_columns(
-        &mut self,
+        &self,
         mut_ref: &mut dyn QTableModelAdapter,
         source_parent: &QModelIndex,
         source_first: i32,
@@ -607,31 +607,31 @@ impl QTableModelProxyRust {
     ) {
         call_cpp_impl!(mut self, mut_ref, base_begin_move_columns(source_parent, source_first, source_last, destination_parent, destination_child))
     }
-    pub fn base_end_move_columns(&mut self, mut_ref: &mut dyn QTableModelAdapter) {
+    pub fn base_end_move_columns(&self, mut_ref: &mut dyn QTableModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_move_columns())
     }
-    pub fn base_begin_move_rows(&mut self, mut_ref: &mut dyn QTableModelAdapter, source_parent: &QModelIndex, source_first: i32, source_last: i32, destination_parent: &QModelIndex, destination_child: i32) {
+    pub fn base_begin_move_rows(&self, mut_ref: &mut dyn QTableModelAdapter, source_parent: &QModelIndex, source_first: i32, source_last: i32, destination_parent: &QModelIndex, destination_child: i32) {
         call_cpp_impl!(mut self, mut_ref, base_begin_move_rows(source_parent, source_first, source_last, destination_parent, destination_child))
     }
-    pub fn base_end_move_rows(&mut self, mut_ref: &mut dyn QTableModelAdapter) {
+    pub fn base_end_move_rows(&self, mut_ref: &mut dyn QTableModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_move_rows())
     }
-    pub fn base_begin_remove_columns(&mut self, mut_ref: &mut dyn QTableModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
+    pub fn base_begin_remove_columns(&self, mut_ref: &mut dyn QTableModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
         call_cpp_impl!(mut self, mut_ref, base_begin_remove_columns(parent, first, last))
     }
-    pub fn base_end_remove_columns(&mut self, mut_ref: &mut dyn QTableModelAdapter) {
+    pub fn base_end_remove_columns(&self, mut_ref: &mut dyn QTableModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_remove_columns())
     }
-    pub fn base_begin_remove_rows(&mut self, mut_ref: &mut dyn QTableModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
+    pub fn base_begin_remove_rows(&self, mut_ref: &mut dyn QTableModelAdapter, parent: &QModelIndex, first: i32, last: i32) {
         call_cpp_impl!(mut self, mut_ref, base_begin_remove_rows(parent, first, last))
     }
-    pub fn base_end_remove_rows(&mut self, mut_ref: &mut dyn QTableModelAdapter) {
+    pub fn base_end_remove_rows(&self, mut_ref: &mut dyn QTableModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_remove_rows())
     }
-    pub fn base_begin_reset_model(&mut self, mut_ref: &mut dyn QTableModelAdapter) {
+    pub fn base_begin_reset_model(&self, mut_ref: &mut dyn QTableModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_begin_reset_model())
     }
-    pub fn base_end_reset_model(&mut self, mut_ref: &mut dyn QTableModelAdapter) {
+    pub fn base_end_reset_model(&self, mut_ref: &mut dyn QTableModelAdapter) {
         call_cpp_impl!(mut self, mut_ref, base_end_reset_model())
     }
     pub fn base_create_index(&self, reference: &dyn QTableModelAdapter, row: i32, column: i32, ptr: usize) -> QModelIndex {
diff --git a/src/qtable_model/proxy_rust_bridge.rs b/src/qtable_model/proxy_rust_bridge.rs
index 7af129c5d..67e0964c6 100644
--- a/src/qtable_model/proxy_rust_bridge.rs
+++ b/src/qtable_model/proxy_rust_bridge.rs
@@ -30,11 +30,11 @@ pub mod ffi {
         # [cxx_name = roleNames]
         fn role_names(&self) -> QHash_i32_QByteArray;
         # [cxx_name = setData]
-        fn set_data(&mut self, index: &QModelIndex, value: &QVariant, role: i32) -> bool;
+        fn set_data(&self, index: &QModelIndex, value: &QVariant, role: i32) -> bool;
         # [cxx_name = removeColumns]
-        fn remove_columns(&mut self, first: i32, count: i32, parent: &QModelIndex) -> bool;
+        fn remove_columns(&self, first: i32, count: i32, parent: &QModelIndex) -> bool;
         # [cxx_name = removeRows]
-        fn remove_rows(&mut self, first: i32, count: i32, parent: &QModelIndex) -> bool;
+        fn remove_rows(&self, first: i32, count: i32, parent: &QModelIndex) -> bool;
         # [cxx_name = sibling]
         fn sibling(&self, row: i32, column: i32, idx: &QModelIndex) -> QModelIndex;
 
